xarray/core/resample.py

```diff
@@ -168,17 +168,18 @@ def interpolate(self, kind: InterpOptions = "linear") -> T_Xarray:
         scipy.interpolate.interp1d
 
         """
-        return self._interpolate(kind=kind)
+        return self._interpolate(kind=kind, **kwargs)
 
-    def _interpolate(self, kind="linear") -> T_Xarray:
+    def _interpolate(self, kind="linear", **kwargs) -> T_Xarray:
         """Apply scipy.interpolate.interp1d along resampling dimension."""
         obj = self._drop_coords()
         (grouper,) = self.groupers
+        kwargs.update({"bounds_error": False})
```

Comment:
```suggestion
        kwargs.setdefault({"bounds_error": False})
```

Now that a user can set `kwargs`, this seems more appropriate?

Reply:
Hi @dcherian , just change to `kwargs.setdefault("bounds_error", False)`. 
I didn't know the `setdefault` method, thank you !